### PR TITLE
BatchNormalization

### PR DESCRIPTION
**Pull request by dfdx**
There are 2 main changes: 

1. Support for multi-output nodes in the graph.
2. BatchNormalization in test mode. 

BatchNormalization in train mode is also implemented, but the result diverges from the one produced with ONNXRunTime. One difference that I fixed is the handling of momentum and corresponding default value: in Flux, which I mostly borrowed the code for norm layers, we use: 
```
running_mean * (1 - momentum) + current_mean * momentum
```
while in ONNX it's the opposite: 
```
running_mean * momentum + current_mean * (1 - mementum)
```
Thus default value is `mtm=0.1` in Flux and `momentum=0.9` in ONNX. But even after fixing it I still can't get equivalent results and can't immediately spot the difference from the implementation described in [the docs](https://github.com/onnx/onnx/blob/master/docs/Operators.md#BatchNormalization).

Anyway, using ONNX in training mode looks more like an advanced features, so for now I just marked the test as broken. In there are no concerns about it from anyone in the next 24 hours, I'm going to merge it and put to the end of backlog.

### CODE REVIEW

**Comment by ToucheSir** [comment]
I assume ONNXRuntime uses the implementation at https://github.com/microsoft/onnxruntime/blob/cee79526fd68b6fb6b09d47dae0d48b47d2f7021/onnxruntime/core/providers/cpu/nn/batch_norm.h by default?

**Comment by dfdx** [comment]
True, but the implementation in `onnxruntime` is 90% optimizations and only 10% logic, so it doesn't make finding the source of the discrepancy any easier :smile: Anyway, it's a question of priorities - whether we want to ensure at every step the implementation is complete or to have a working propotype suitable for most typical use cases first. 

One quick but safe option may be to throw an error if a user tries to use BatchNormalization in training mode which we aren't 100% sure about.

**Comment by dfdx** [comment]
It turns out I still passed the wrong default value for momentum in one place. After fixing it results are consistent between ONNXRunTime and our implementation in both - training and testing mode.
